#  Pandas project  - Jacobo Pintos outputs

Con la información obtenida de ataques de tiburón en USA entre 1950 y 2016 vamos a responder a varias preguntas que son:

    1. ¿En que zona de EEUU es donde más ataques de tiburón se registraron? 
    2. ¿Que porcentaje de estos ataques son mortales?
    3. ¿Existe alguna diferencia en cuanto al sexo?

In [1]:
import pandas as pd

In [2]:
sharks_info = pd.read_csv('sharks.csv', encoding='ISO/8859/1')

In [3]:
sharks_info.head()

Year        Type Country          US zone     Area     Sex     Age  \
0  2016  Unprovoked     USA         SE coast  Florida    Male      16   
1  2016  Unprovoked     USA         SE coast  Florida    Male      36   
2  2016  Unprovoked     USA         SE coast  Florida    Male      43   
3  2016  Unprovoked     USA         SE coast  Florida    Male  Uknown   
4  2016  Unprovoked     USA  Pacific Islands   Hawaii  Female      51   

  Fatal Attack  
0           No  
1           No  
2           No  
3           No  
4           No

## 1. ¿En que zona de EEUU es donde más ataques de tiburón se registraron? 

In [4]:
sharks_info['US zone'].value_counts()

SE coast              1101
SW coast               224
Pacific Islands        202
NE coast                46
NW coast                29
Caribean Islands        11
US Central               2
Philippine Islands       1
Name: US zone, dtype: int64

En la zona del South East de USA se registraron un total de 1101 ataques. 

In [5]:
sharks_info['US zone'].value_counts(1)*100

SE coast              68.131188
SW coast              13.861386
Pacific Islands       12.500000
NE coast               2.846535
NW coast               1.794554
Caribean Islands       0.680693
US Central             0.123762
Philippine Islands     0.061881
Name: US zone, dtype: float64

Estos 1101 ataques supusieron el 68% de todos los ataques en USA desde 1950. Si vemos que Area fue la que registró mas ataques.

In [6]:
SE_coast_info = sharks_info.drop(sharks_info[sharks_info['US zone']!='SE coast'].index)
SE_coast_info['Area'].value_counts(1)*100

Florida           77.384196
South Carolina     8.900999
North Carolina     6.539510
Texas              4.541326
Alabama            0.999092
Georgia            0.726612
Louisiana          0.635786
Mississippi        0.272480
Name: Area, dtype: float64

Podemos ver como Florida es la zona de mayor influencia con un 77% de los ataques registrados. Podemos ver el peso porcentual de Florida vs el resto de Areas de USA.

In [7]:
sharks_info['Area'].value_counts(1).nlargest(5)*100

Florida           52.722772
California        13.799505
Hawaii            12.252475
South Carolina     6.064356
North Carolina     4.455446
Name: Area, dtype: float64

Florida supone el 53% de los ataques de tiburón seguido en el top 3 de California con un 14% y Hawaii con un 12%.

## 2 ¿Que porcentaje los ataques registrados fueron mortales? ¿En que zona hay mayor indice de mortalidad por ataque?

In [8]:
fatal_info = sharks_info.drop(sharks_info[sharks_info['Fatal Attack']!='Yes'].index)

In [9]:
fatal_info['US zone'].value_counts()

Pacific Islands     25
SE coast            23
SW coast            15
Caribean Islands     4
NE coast             2
Name: US zone, dtype: int64

In [10]:
fatal_info2 = fatal_info.drop(fatal_info[fatal_info['US zone']!='Pacific Islands'].index)

In [11]:
fatal_info2['Area'].value_counts()

Hawaii    25
Name: Area, dtype: int64

In [12]:
fatal_info['US zone'].value_counts()/sharks_info['US zone'].value_counts()*100

Caribean Islands      36.363636
NE coast               4.347826
NW coast                    NaN
Pacific Islands       12.376238
Philippine Islands          NaN
SE coast               2.089010
SW coast               6.696429
US Central                  NaN
Name: US zone, dtype: float64

La probabilidad de muerte por ataque de tiburón en el South East de USA es de un 2%, en cambio otras geografías relevantes como el South West tiene una probabilidad de casi el 7%. 

El porcentaje de mortalidad más alto lo tienen las Caribean Islands donde el 36% de los atacados por tiburón resultaron muertos (4/11). Con mayor relevancia debido al número de ataques nos encontramos con el caso de las islas del Pacifico donde 25 de los 202 casos registrados acabaron en muerte.

## 3. ¿Existe alguna diferencia entre los ataques provocados y no provocados en cuanto al sexo?

In [13]:
gender_analysis = sharks_info.drop(sharks_info[sharks_info['Type']=='Boat'].index)
gender_analysis = sharks_info.drop(sharks_info[sharks_info['Sex']=='Uknown'].index)

In [14]:
Provoked_attacks = gender_analysis.drop(gender_analysis[gender_analysis['Type']=='Unprovoked'].index)

In [15]:
Provoked_attacks['Sex'].value_counts(1)*100

Male      93.710692
Female     6.289308
Name: Sex, dtype: float64

In [16]:
Unprovoked_attaks = gender_analysis.drop(gender_analysis[gender_analysis['Type']=='Provoked'].index)

In [17]:
Unprovoked_attaks['Sex'].value_counts(1)*100

Male      81.889205
Female    18.110795
Name: Sex, dtype: float64

Dejando a un lado el % de desconocidos, los hombres representan el 94% de los ataques de tiburones provocados mientras que las mujeres solo un 6%, en cambio si vemos los no provocados este porcentaje de mujeres se triplica hasta el 18% dejando el de hombres en un 82% lo que supone un cambio muy significativo.